Expose torch.distributed.is_available() API

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37021 Expose torch.distributed.is_available() API**

Differential Revision: [D21164318](https://our.internmc.facebook.com/intern/diff/D21164318)

Commit:
[WIP] Expose torch.distributed.is_available() API

[ghstack-poisoned]

mrshenli commented:
<img width="868" alt="Screen Shot 2020-04-21 at 5 20 39 PM" src="https://user-images.githubusercontent.com/16999635/79915167-7bc52d00-83f4-11ea-94d7-77760adfd5df.png">

Commit:
Update on "Expose torch.distributed.is_available() API"




[ghstack-poisoned]

mrshenli commented:
This is trying to address comments from this [forum post](https://discuss.pytorch.org/t/module-torch-distributed-has-no-attribute-is-initialized/55391/6)

rohan-varma commented on torch/distributed/__init__.py:
```diff
@@ -4,6 +4,12 @@
+    Returns ``True`` if the distributed package is available. Otherwise,
+    ``torch.distributed`` does not expose any other APIs. Currently,
+    ``torch.distributed`` is available on Linux and MacOS. Set
+    ``USE_DISTRIBUTED=1`` at compile time to enable it.
```
It might be helpful to specify that this is for building from source?
Also, what's the current default?

mrshenli replied:
good point! will update

current default: USE_DISTRIBUTED=1 on Linux and USE_DISTRIBUTED=0 on MacOS, let me include those

Commit:
Update on "Expose torch.distributed.is_available() API"


Differential Revision: [D21164318](https://our.internmc.facebook.com/intern/diff/D21164318)

[ghstack-poisoned]
